modules/nixvim/nixvim.nix

```diff
@@ -63,80 +63,81 @@ in
     })
   ];
 
-  config =
-    lib.mkIf (config.stylix.enable && cfg.enable && (config.programs ? nixvim))
-      (
-        lib.optionalAttrs (builtins.hasAttr "nixvim" options.programs) (
-          lib.mkMerge [
-            (lib.mkIf (cfg.plugin == "base16-nvim") {
-              programs.nixvim.colorschemes.base16 = {
-                enable = true;
+  config = {
+    programs = lib.mkIf (config.stylix.enable && cfg.enable) (
+      lib.optionalAttrs (options.programs ? nixvim) {
+        nixvim = config.lib.stylix.nixvim.config;
+      }
+    );
+    lib.stylix.nixvim.config = lib.mkMerge [
+      (lib.mkIf (cfg.plugin == "base16-nvim") {
```

Comment:
I may be mis-remembering, but I think this might fail to merge correctly because `config.lib` is an `attrsOf attrs` type, and the element-type (`attrs`) doesn't support nested merging IIRC.

Again, this may be simpler to ditch the `lib` namespace idea and go back to declaring a dedicated read-only option somewhere. Thoughts?